### PR TITLE
Fix CORS errors in FLINT Example REST API

### LINKED ISSUE #19: Add support for Cross-Origin Resource Sharing (CORS) in the FLINT Example REST API
### The Problem:

I am working on a proof of concept of a user interface for FLINT Example REST API.

I am trying to connect the [basic UI](https://github.com/waridrox/basic-ui) made by @waridrox to the FLINT Example but the request gets blocked by the Cross-Origin Resource Sharing (CORS) policy.  
[Please go through this great document on MDN to understand the CORS policy and its implications. ](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors)

This is the console log:
```
xhr.js?ec6c:177 GET http://127.0.0.1:8080/spec net::ERR_FAILED
dispatchXhrRequest @ xhr.js?ec6c:177
xhrAdapter @ xhr.js?ec6c:13
dispatchRequest @ dispatchRequest.js?c4bb:52
Promise.then (async)
request @ Axios.js?5e65:61
Axios.<computed> @ Axios.js?5e65:76
wrap @ bind.js?24ff:9
apiRoute_spec @ App.vue?26cd:53
click @ App.vue?edc5:27
invokeWithErrorHandling @ vue.esm.js?efeb:1872
invoker @ vue.esm.js?efeb:2197
original._wrapper @ vue.esm.js?efeb:7609
:8000/#/:1 Access to XMLHttpRequest at 'http://127.0.0.1:8080/spec' from origin 'http://127.0.0.1:8000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
App.vue?26cd:55 Error: Network Error
    at createError (createError.js?16d0:16)
    at XMLHttpRequest.handleError (xhr.js?ec6c:84)
```

image

### Proposed Solution:

This issue can be solved in two ways:

Both the solutions involve enabling CORS, but I do not know about the severity of the security risks we may encounter.
1.  Client side: by using a [Chrome extension](https://chrome.google.com/webstore/detail/allow-cors-access-control/lhobafahddgcelffkeicbaginigeejlf?hl=en), but this will be too much work for the average user, which defeats the purpose of building a user interface for the FLINT.
2.  Server side: by making a few changes in the API itself. We can use the [Flask-CORS library](https://flask-cors.readthedocs.io/en/latest/) with the API to enable CORS. In this way the user would not have to make any environment changes on their machine.

This is how the console looks after hitting the _`/spec`_ endpoint from @waridrox 's UI via Axios with Flask-CORS:

image

@Tlazypanda @arnav-t @aornugent @mfellows @abhi211199 
As the user would be working on the local Docker environment as well as the FLINT Cloud environment, is the solution sustainable from a security standpoint? Please give your views on this! Thank you very much!

cc: @HarshCasper @waridrox

### PR DESCRIPTION
**Pull request by shubhamkarande13**
Adds support to enable CORS in FLINT Example REST API and fixes #19.

### CODE REVIEW

**Comment by aornugent** [comment]
Looks good to me @shubhamkarande13. Let's give @Tlazypanda a little more time to check this but I think you can merge it straight after.

File: rest_api_flint.example/app.py [hunk 1]
```diff
@@ -7,10 +7,14 @@
 from datetime import datetime
 from flask_swagger import swagger
 from flask_swagger_ui import get_swaggerui_blueprint
+from flask_cors import CORS
 
 app = Flask(__name__)
+app = Flask(__name__)
+CORS(app)
```
Comment (Tlazypanda) [inline rest_api_flint.example/app.py:14]: Can we add the origins like this for local host and other deployed links?

Example:
`cors = CORS(app, origins=['http://localhost:5000',r'^https://.+example.com$'])`

**Review by Tlazypanda — COMMENTED** [review]
Rest all looks great 🎉

File: rest_api_flint.example/app.py (hunk 1 above)
Comment (shubhamkarande13) [reply rest_api_flint.example/app.py:14]: I am not sure how would the two containers (API and the UI) will talk to each other, and so I can not fix the domains for now. Can we do this once the IP for the UI container is fixed? 😅
`CORS(app)` enables CORS on all domains I suppose.

Comment (aornugent) [reply rest_api_flint.example/app.py:14]: Start with localhost for now and we'll add more origins as needed. I think this approach forces people to 'opt-in' to CORS when setting up their domain, which is better practice than enabling it by default and having people 'opt-out'.

Comment (shubhamkarande13) [reply rest_api_flint.example/app.py]: Done!

**Comment by aornugent** [comment]
I think you can merge this @shubhamkarande13 you've made the requested changes.

**Comment by shubhamkarande13** [comment]
Umm, @aornugent I do not have the sufficient privileges to merge this PR.  😅

**Review by Tlazypanda — APPROVED** [review]
Approved 🎉

**Comment by aornugent** [comment]
I've just finished setting up a FLINTcloud Github team.

**Pull request merged**